application/forms/Authentication/LoginForm.php

```diff
@@ -150,4 +169,42 @@ public function onRequest()
             ));
         }
     }
+
+    /**
+     * This function encrypt username and password,
+     * change these into binary code
+     * @param $username
+     * @param $password
+     * @return array
+     */
+    public function encryptRememberMeData($username, $password)
```

Comment:
It's good that you made a function for the encryption but we have to go a step further because we also have to save the created keys in oder to store them into the database. And of course, we have to test the encryption/decryption.

I opt for creating the class `Icinga\Crypt\RSA` like this:

```php

namespace Icinga\Crypt;

class RSA
{
    public static function keygen();

    public function getPublicKey();

    public function getPrivateKey();

    public function decrypt(...$data);

    public function decryptFromBase64(...$data);

    public function encrypt(...$data);

    public function encryptToBase64(...$data);

    public function loadKey(...$key);
}
```

`RSA::keygen()` generates a new private key and public and returns it as array with the private as first element and the public key as second. The default key size should be 4096 bits.`RSA::loadKey()` should auto-detected the key type. I don't want to specify whether I am loading a private or public key but `loadKey()` should handle this. Please find out how we could do this. Every other method should be self-explanatory.

Usage would be as follows:

```php

// Encryption
$rsa = new RSA();
$rsa->loadKey(...RSA::keygen());

$data = $rsa->encryptToBase64($username, $password);
$data[] = base64_encode($rsa->getPublicKey());
$data = implode('|', $data);

// Decryption
$data = load_cookie();
$data = explode('|', $data);
$publicKey = base64_decode(array_pop($data));

$rsa = new RSA();
$rsa->loadKey($publicKey, load_private_key_from_database($publicKey));

list($username, $password) = $rsa->decryptFromBase64(...$data);